packages/block-library/src/preformatted/editor.scss

```diff
@@ -0,0 +1,4 @@
+.wp-block-preformatted {
+	overflow-x: auto;
+	white-space: pre !important;
```

Comment:
Since this rule also exists in `style.scss`, we are so close to not needing any editor style at all for this block. And we should be avoiding !important anyway, if at all possible. I dug in to see why the !important was necessary, and it appears that `white-space: pre-wrap;` is applied as an inline style on the preformatted block:

<img width="619" alt="Screenshot 2020-12-09 at 08 20 30" src="https://user-images.githubusercontent.com/1204802/101597854-91c14580-39f7-11eb-8c28-a825bd58ec6e.png">

Why does that exist? I'm not sure, but from digging at the code, it looks to be related to this PR: #18656. Or possibly #17779. Specifically there's some commentary in https://github.com/WordPress/gutenberg/pull/17779/files#diff-7df6f5253ccec2f2e6d1bb22e81f87a4b6bccd2604463eeecded163aa27fd333R70 which seems relevant.

It seems we need to figure out why that output

Reply:
The `pre-wrap` inline style is coming from the `<Richtext>` component, and its value is defined here: https://github.com/WordPress/gutenberg/blob/master/packages/rich-text/src/component/index.js#L82

The value is set on the `whiteSpace` variable, which is then added to `defaultStyle`, which subsequently makes it to the component's inline styles via the `style` prop here: https://github.com/WordPress/gutenberg/blob/master/packages/rich-text/src/component/index.js#L1075

Reply:
According to the inspector, I can see that besides the `pre-wrap` inline style, there's the `pre-wrap` style attribute coming from the Block's `style.scss` stylesheet. If neither the inline style nor the block style are present, then, the `<pre>` element will get its default style from the browser defaults.

This makes me think we can live with the inline style, since the `<RichText>` component is using the style attribute to guarantee its proper display (my assumption). Assuming a future scenario where the `<RichText>` component removes the `pre-wrap` inline style, then we'll still be covered by the `pre-wrap` style available in the preformatted block's `style.scss`.

Reply:
Why not just remove the inline style?

Reply:
That's a quite considerable change. The `<RichText>` component uses it by default, and it is used in many places. Changing this may change the overall behavior of the `<RichText>` component.